src/map/script.c

```diff
@@ -13546,6 +13575,200 @@ BUILDIN(getiteminfo)
+	
+	return true;
+}
+
+/**
+ * Set an equipment's option value.
+ * *setequipoption(<equip_index>,<slot>,<opt_index>,<value>);
+ *
+ * @param equip_index     as the inventory index of the equipment.
+ * @param slot            as the slot of the item option (1 to MAX_ITEM_OPTIONS)
+ * @param opt_index       as the index of the option available as "Id" in db/item_options.conf.
+ * @param value           as the value of the option type.
+ *                        For IT_OPT_INDEX see "Name" in item_options.conf
+ *                        For IT_OPT_VALUE, the value of the script bonus.
+ * @return 0 on failure, 1 on success.
+ */
+BUILDIN(setequipoption)
+{
+	int equip_index = script_getnum(st, 2);
+	int slot = script_getnum(st, 4);
+	int opt_index = script_getnum(st, 3);
+	int value = script_getnum(st, 5);
+	int i = -1;
+	
+	struct map_session_data *sd = script->rid2sd(st);
+	struct item_option *ito = NULL;
+	
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		ShowError("buildin_setequipoption: Player not attached!\n");
+		return false;
+	}
+
+	if (equip_index < 0) {
+		script_pushint(st, 0);
+		ShowError("buildin_setequipoption: Invalid equipment index %d provided.\n", equip_index);
+		return false;
+	}
+
+	if (slot <= 0 || slot > MAX_ITEM_OPTIONS) {
+		script_pushint(st, 0);
+		ShowError("buildin_setequipoption: Invalid option index %d (Min: 1, Max: %d) provided.\n", slot, MAX_ITEM_OPTIONS);
+		return false;
+	}
+
+	if (equip_index > 0 && equip_index <= ARRAYLENGTH(script->equip)) {
+		if ((i = pc->checkequip(sd, script->equip[equip_index - 1])) == -1) {
+			ShowError("buildin_setequipoptioninfo: No equipment is equipped in the given index %d.\n", equip_index);
+			script_pushint(st, 0);
+			return false;
+		}
+	} else {
+		ShowError("buildin_setequipoptioninfo: Invalid equipment index %d provided.\n", equip_index);
+		script_pushint(st, 0);
+		return false;
+	}
+	
+	if (sd->status.inventory[i].nameid) {
+		
+		if ((ito = itemdb->option_exists(opt_index)) == NULL) {
+			script_pushint(st, 0);
+			ShowError("buildin_setequipotion: Option index %d does not exist!\n", opt_index);
+			return false;
+		} else if (value < -INT16_MAX || value > INT16_MAX) {
+			script_pushint(st, 0);
+			ShowError("buildin_setequipotion: Option value %d exceeds maximum limit (%d to %d) for type!\n", value, -INT16_MAX, INT16_MAX);
+			return false;
+		}
+		/* Add Option Index */
+		sd->status.inventory[i].option[slot-1].index = ito->index;
+		/* Add Option Value */
+		sd->status.inventory[i].option[slot-1].value = value;
+		
+		/* Unequip and simulate deletion of the item. */
+		pc->unequipitem(sd, i, PCUNEQUIPITEM_FORCE); // status calc will happen in pc->equipitem() below
+		clif->refine(sd->fd, 0, i, sd->status.inventory[i].refine); // notify client of a refine.
+		clif->delitem(sd, i, 1, DELITEM_MATERIALCHANGE); // notify client to simulate item deletion.
+		/* Log the Item */
+		logs->pick_pc(sd, LOG_TYPE_SCRIPT, 1, &sd->status.inventory[i], sd->inventory_data[i]);
```

Comment:
Logs should happen in pairs, there should be a matching entry with `-1` before (see `BUILDIN(successrefitem)` for reference)